### PR TITLE
M sma 64 master

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: test/Smart-Account-Factory.specs.ts [hunk 1]
```diff
@@ -89,23 +89,200 @@ describe("Smart Account Factory", async () => {
...
+        const stakeAmount = ethers.utils.parseEther("1.234256");
+        await smartAccountFactory.addStake(entryPoint.address, 600, {
+          value: stakeAmount,
+        });
+        // unstake
+        const tx = await smartAccountFactory.unlockStake(entryPoint.address);
+        await expect(tx).to.emit(entryPoint, "StakeUnlocked");
+      });
+
+      // positive cases
```
Comment (filmakarov) [inline test/Smart-Account-Factory.specs.ts:182]: it is not the positive case

File: test/Smart-Account-Factory.specs.ts [hunk 2]
```diff
@@ -89,23 +89,200 @@ describe("Smart Account Factory", async () => {
...
+        // staking first
+        const stakeAmount = ethers.utils.parseEther("1.234256");
+        await smartAccountFactory.addStake(entryPoint.address, 1, {
+          value: stakeAmount,
+        });
+        // waiting for additional blocks
+        const blocksToWait = 10;
+        const currentBlockNumber = await ethers.provider.getBlockNumber();
+        const targetBlockNumber = currentBlockNumber + blocksToWait;
+        while ((await ethers.provider.getBlockNumber()) < targetBlockNumber) {
```
Comment (filmakarov) [inline test/Smart-Account-Factory.specs.ts:222]: I suggest using `evm_increaseTime` instead

File: test/Smart-Account-Factory.specs.ts [hunk 3]
```diff
@@ -89,23 +89,200 @@ describe("Smart Account Factory", async () => {
...
+          await ethers.provider.send("evm_mine", []);
+        }
+        // withdrawing
+        await expect(
+          smartAccountFactory.withdrawStake(entryPoint.address, alice.address)
+        ).to.be.revertedWith("must call unlockStake() first");
+      });
 
-    // not owner cannot add, unlock, withdraw
+      it("reverts when calling before stake withdrawal", async () => {
```
Comment (filmakarov) [inline test/Smart-Account-Factory.specs.ts:264]: what is this test about?

File: test/Smart-Account-Factory.specs.ts
Comment (filmakarov) [inline test/Smart-Account-Factory.specs.ts:1]: I suggest adding test for this require
```
require(epAddress != address(0), "Invalid EP address")
```
as well as tests to what happens if address provided is not a valid EP

**Review by filmakarov — CHANGES_REQUESTED** [review]

File: test/Smart-Account-Factory.specs.ts
Comment (proxima424) [reply test/Smart-Account-Factory.specs.ts:1]: Yes, will make the following changes

File: test/Smart-Account-Factory.specs.ts (hunk 3 above)
Comment (proxima424) [reply test/Smart-Account-Factory.specs.ts:264]: It is for [this](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/core/StakeManager.sol#L120) require check in StakeManager.sol. 
Before calling withdrawStake(), it checks whether the withdrawTime is due or not.

File: test/Smart-Account-Factory.specs.ts (hunk 1 above)
Comment (proxima424) [reply test/Smart-Account-Factory.specs.ts:182]: yes, made the changes.

**Comment by filmakarov** [comment]
@proxima424 @2Proxima4 What's the status here?

**Comment by Aboudjem** [comment]
Closing this PR as there hasn't been any recent update and lack of necessary context, please refer to our guidelines👍:

[Branch Naming Convention](https://www.notion.so/Branch-Naming-Convention-ecb3dd197a1b4e7f83de2a0efdc03233?pvs=21)
[Git Workflow: From Feature Dev to Prod Release](https://www.notion.so/Git-Workflow-From-Feature-Dev-to-Prod-Release-abd9c27ad487443b9aeb9e3e243cc579?pvs=21)
[Gitmoji Commit Best Practices](https://www.notion.so/Gitmoji-Commit-Best-Practices-e495d99068cf4db58de8b4de122d3f95?pvs=21)
[Pull Requests (PRs)](https://www.notion.so/Pull-Requests-PRs-d64795d49f8d491caba4df7f4f13507f?pvs=21).

**Pull request closed without merging**